apache/superset
# feat: Stop async queries

### SUMMARY
Currently async queries executed in Chart view cannot be stopped. 

To stop a running query in the backend the following changes were required:
- use existing `job_id` (a random UUID that is returned to frontend) as celery task id
- endpoint `DELETE /api/v1/chart/async_job/<string:job_id>` to terminate/revoke Celery task with a given task id
- close db cursors for engines that use it to cancel a running query (e.g. Trion client)

In frontend we need to call endpoint to stop an async query when:
- AbortController communicates an abort signal ('Stop' button was clicked)
- Chart component will unmount
- beforeunload event is fired

Frontend keeps job id returned by backend in chart state. 

### TESTING INSTRUCTIONS
Enable and configure `GLOBAL_ASYNC_QUERIES` feature. Next go to Chart view, execute a long running query, and verify that Celery task is terminated and query in database is cancelled when:
- 'Stop' button is clicked
- Chart view is changed (e.g. click 'Home' tab)
- browser tab is closed
- browser tab is refreshed


### ADDITIONAL INFORMATION


- [x] Has associated issue: https://github.com/apache/superset/issues/24890
- [x] Required feature flags: `GLOBAL_ASYNC_QUERIES: True`
- [x] Introduces new feature or API - adds endpoint `DELETE /api/v1/chart/async_job/<string:job_id>`
